backend/RoomPlanner/room-planner/src/main/java/edu/roomplanner/service/impl/ReservationServiceImpl.java

```diff
@@ -88,4 +100,72 @@ private InvalidReservationException createInvalidReservationException(List<Strin
         return new InvalidReservationException(validatorErrors.toString().substring(0, validatorErrors.length() - 4));
     }
 
+    @Override
+    public ReservationDto updateReservation(Long id, ReservationDto reservationDto) {
+
+        ReservationDto copyDto = ReservationDtoBuilder.builder().withEndDate(Calendar.getInstance()).withStartDate(Calendar.getInstance()).build();
+        copyDto.getStartDate().setTime((Date) reservationDto.getStartDate().getTime().clone());
+        copyDto.getEndDate().setTime((Date) reservationDto.getEndDate().getTime().clone());
+
+        copyDto.getStartDate().setTime(conversionToGmt(reservationDto.getStartDate().getTime()));
+        copyDto.getEndDate().setTime(conversionToGmt(reservationDto.getEndDate().getTime()));
+
+        String userEmail = tokenParserService.getEmailFromToken();
+        if (!verifyParameters(userEmail, reservationDto, id)) {
+            return null;
+        }
+        ReservationEntity reservationEntity = reservationRepository.findById(id).get();
+        Long userId = userRepository.findByEmail(userEmail).get().getId();
+        ValidationResult result = startEndDateValidator.validate(getReservation(copyDto));
+        if (result.getError() != null) {
+            return null;
+        }
+        if (userId.equals(reservationEntity.getPerson().getId())) {
+            reservationEntity.setStartDate(reservationDto.getStartDate());
```

Comment:
You could break this portion of code in other function to make the code more clean